dbt/adapters/spark/impl.py

```diff
@@ -133,44 +135,114 @@ def add_schema_to_cache(self, schema) -> str:
         # so jinja doesn't render things
         return ""
 
-    def list_relations_without_caching(
-        self, schema_relation: SparkRelation
-    ) -> List[SparkRelation]:
-        kwargs = {"schema_relation": schema_relation}
-        try:
-            results = self.execute_macro(LIST_RELATIONS_MACRO_NAME, kwargs=kwargs)
-        except dbt.exceptions.DbtRuntimeError as e:
-            errmsg = getattr(e, "msg", "")
-            if f"Database '{schema_relation}' not found" in errmsg:
-                return []
-            else:
-                description = "Error while retrieving information about"
-                logger.debug(f"{description} {schema_relation}: {e.msg}")
-                return []
+    def _get_relation_information(self, row: agate.Row) -> Tuple[str, str, str]:
+        """relation info was fetched with SHOW TABLES EXTENDED"""
+        if len(row) == 4:
+            _schema, name, _, information = row
+            return _schema, name, information
+        else:
+            raise dbt.exceptions.DbtRuntimeError(
+                f'Invalid value from "show tables extended ...", got {len(row)} values, expected 4'
+            )
 
-        relations = []
-        for row in results:
-            if len(row) != 4:
-                raise dbt.exceptions.DbtRuntimeError(
-                    f'Invalid value from "show table extended ...", '
-                    f"got {len(row)} values, expected 4"
+    def _get_relation_information_using_describe(self, row: agate.Row) -> Tuple[str, str, str]:
+        """Relation info fetched using SHOW TABLES and an auxiliary DESCRIBE statement"""
+        if len(row) == 3:
+            _schema, name, _ = row
+
+            table_name = f"{_schema}.{name}"
+            information = ""
+            try:
+                table_results = self.execute_macro(
+                    DESCRIBE_TABLE_EXTENDED_MACRO_NAME, kwargs={"table_name": table_name}
                 )
-            _schema, name, _, information = row
-            rel_type = RelationType.View if "Type: VIEW" in information else RelationType.Table
-            is_delta = "Provider: delta" in information
-            is_hudi = "Provider: hudi" in information
-            relation: BaseRelation = self.Relation.create(
+
+                for info_row in table_results:
+                    info_type, info_value, _ = info_row
+                    if not info_type.startswith("#"):
+                        information += f"{info_type}: {info_value}\n"
+            except dbt.exceptions.DbtRuntimeError as e:
+                logger.debug(f"Error while retrieving information about {table_name}: {e.msg}")
+
+            return _schema, name, information
```

Comment:
This bears some explaining: The way the `parse_information` method was written meant returning an empty string was returned when something went wrong:
``` python
def parse_information(self, table_name: str) -> str:
        information = ""
        try:
            table_results = self.execute_macro(
                DESCRIBE_TABLE_EXTENDED_MACRO_NAME, kwargs={"table_name": table_name}
            )
        except dbt.exceptions.DbtRuntimeError as e:
            logger.debug(f"Error while retrieving information about {table_name}: {e.msg}")
            return ""
        for info_row in table_results:
            info_type, info_value, _ = info_row
            if not info_type.startswith("#"):
                information += f"{info_type}: {info_value}\n"
        return information
```
That behavior continues here as I've written it. The try-catch is there to log what happened before exiting. The information string being empty is an indication that there was a bug in this process (unless we want to explicitly say that)